pillar/cni.sls

```diff
@@ -16,6 +16,9 @@ flannel:
 # 7 - Display HTTP request headers.
 # 8 - Display HTTP request contents.
   log_level:      '2'
+# cilium configuration
+cilium:
+  image:          '{{ salt.caasp_registry.base_image_url() }}/cilium:1.2.1'
```

Comment:
@inercia was suggesting to not hardcode the image version here but to add a pillar in [params.sls](https://github.com/kubic-project/salt/blob/master/pillar/params.sls)

See https://github.com/kubic-project/salt/pull/747#discussion_r258829813

Reply:
Nice suggestion, but Salt has a terrible limitation: you cannot use pillars when defining pillars. So we must define the `image` here or in the `params.sls`, but it cannot be built as a mix.

Reply:
I think you cannot (or should not) use `salt.caasp_registry.base_image_url()` here. You could use it in the places where you get the `cilium:image` pillar, and specify here just the `name:tag`, ie:

```YAML
cilium:
  image:          'cilium:1.2.1'
```

Reply:
@lcavajani @inercia 's suggestion make sense for that PR as we don't want to hardcode container image in a jinja file. We have pillar/cni.sls so it doesn't make sense to spread cni reference to pillar/param.sls